### DISCUSSION TITLE
ATTiny85 programming on an USBTinyISP (Spark clone Tiny ISP) errors on COM5

### DISCUSSION THREAD

**Discussion by johndsummers** [SpenceKonde/ATTinyCore#602] [Support / Q & A]
Hello community,

 I have been digging through the web trying to find a solution for the problem I am having. 
I have a SparkFun Tiny USB ISP AVR programmer. I saw others in this discussion forum use it to program their chips. 
I can load a bootloader on the chip just fine. I have put the Optiboot core on the chip. 
When when I go to upload a sketch, I get this error: 
        System wide configuration file is "C:\Users\me\AppData\Local\Arduino15\packages\ATTinyCore\hardware\avr\1.5.2/avrdude.conf"

         Using Port                    : COM5
An error occurred while uploading the sketch
         Using Programmer              : arduino
         Overriding Baud Rate          : 19200
         Setting bit clk period        : 5.0
avrdude: ser_open(): can't open device "\\.\COM5": The system cannot find the file specified.

I have tried all the variations of the USBTinyISP in the list and none work. 

Settings:
Board: ATTiny85/45 Optiboot
Chip: ATTiny85
Clock source: 8mhz internal
Timer1 CPU Frequency
LTO: Enabled
Millis(): Enabled
BOD: Disabled
No port selected

Running Arduino IDE 1.8.13
I could use your help with this, please!!

**Comment by hmeijdam** [comment]
Which programmer do you select in your IDE and do you use the "Upload"  (CTRTL+U)  or "Upload with programmer"  (CTRL+SHIFT+U)

I suspect you do the first as your error message shows " Using Programmer : arduino"

**Reply by johndsummers** [reply]
Thank you for the response. 
I used the 2 variations of USBTinyISP (ATTinyCore) Slow and then the ATTinyCore Fast.

Edit: It worked using the Upload with programmer key combo. Thank you hmeijdam very much!!!

You ROCK!!!

**Comment by hmeijdam** [comment]
And you don't need to remember the key combo.
![image](https://user-images.githubusercontent.com/9595465/131566887-4dedfd22-5cd9-4a3e-8d3e-4ed9a2c09d62.png)

**Reply by Erik84750** [reply]
Why not use [AVRDUDESS ](https://blog.zakkemble.net/avrdudess-a-gui-for-avrdude/)?
It provides fuse writing options, and a very usefull user interface.

**Reply by hmeijdam** [reply]
While I agree that Avrdudess is a very handy GUI as wrapper around avrdude commandline, it would have required more explaining how to upload a sketch (export compiled binary, upload via avrdudess...), compared to suggesting to use the correct upload command "Upload with programmer" (CTRL+SHIFT+U)
